src/resolution_functions/models/pychop.py

```diff
@@ -491,6 +566,7 @@ def _get_chop_times(cls,
         chop_times = []
 
         # first we optimise on the main Ei
+        # TODO: Calculate only the first and last choppers (only those used upstream)
```

Comment:
Is this still TODO, and would it be better tracked as an Issue? TODOs in code bases tend to be forgotten unless there is an automated procedure flagging them.